src/app/app-interactive-map/app-interactive-map.component.ts

```diff
@@ -125,13 +127,13 @@ export class AppInteractiveMapComponent implements OnInit, AfterViewInit {
         // opening the dialog throws ExpressionChangedAfterItHasBeenCheckedError
         // See https://github.com/angular/material2/issues/5268#issuecomment-416686390
         // setTimeout(() => ...., 0);
-        if (!this.dialog.openDialogs.find((dialog) => dialog.id === 'loader')) {
+        if (!this.dialog.openDialogs.find((dialog) => dialog.id === 'loader') && this.router.url === '/interactiveMap') {
```

Comment:
I guess this is needed since we don't unsubscribe when leaving the component? Still feels pretty hacky though.